src/magnify/apps/core/utils.py

```diff
@@ -44,49 +50,70 @@ def get_nth_week_number(original_date):
+    if not room.configuration["screenSharingEnabled"]:
+        sources.remove("screen_share")
+        sources.remove("screen_share_audio")
+    return sources
+
+
+def create_video_grants(room: string, is_admin=False, is_temp_room=True):
+    """Creates video grants given room and user permission"""
+
+    grants = api.VideoGrants(room_join=True, room=room, can_publish=False, can_subscribe=False, room_admin=is_admin,
+                             can_update_own_metadata=True, can_publish_sources=["camera", "microphone", "screen_share", "screen_share_audio"])
+
+    if is_temp_room:
+        grants = api.VideoGrants(room_join=True, room=room, can_publish=True, can_subscribe=True, room_admin=is_admin,
+                                 can_update_own_metadata=True, can_publish_sources=["camera", "microphone", "screen_share", "screen_share_audio"])
+        return grants
+
+    try:
+        roomData = models.Room.objects.get(id=uuid.UUID(room))
+
+        chat_enabled = roomData.configuration['enableLobbyChat']
+        grants = api.VideoGrants(room_join=True, room=room, can_publish=False, can_subscribe=False, room_admin=is_admin,
+                                 can_update_own_metadata=True, can_publish_sources=get_publish_sources(roomData, is_admin), can_publish_data=chat_enabled)
+        if is_admin or not roomData.configuration["waitingRoomEnabled"]:
+            grants = api.VideoGrants(room_join=True, room=room, can_publish=True, can_subscribe=True, room_admin=is_admin,
+                                     can_update_own_metadata=True, can_publish_sources=get_publish_sources(roomData, is_admin), can_publish_data=chat_enabled)
+    finally:
+        return grants
+
+
+def create_livekit_token(identity, username, room, is_admin=False, is_temp_room=True):
+    """Create the token so that it contains each information LiveKit requires"""
     expiration_seconds = int(
-        settings.JITSI_CONFIGURATION["jitsi_token_expiration_seconds"]
+        settings.LIVEKIT_CONFIGURATION["livekit_token_expiration_seconds"]
     )
-    token_payload = {
-        "exp": timezone.now() + timedelta(seconds=expiration_seconds),
-        "iat": timezone.now(),
-        "moderator": is_admin or user.is_staff,
-        "aud": "jitsi",
-        "iss": settings.JITSI_CONFIGURATION["jitsi_app_id"],
-        "sub": settings.JITSI_CONFIGURATION["jitsi_xmpp_domain"],
-        "room": room,
-    }
-
-    jitsi_user = {
-        "avatar": settings.JITSI_CONFIGURATION.get("jitsi_guest_avatar"),
-        "name": user.username
-        if user.is_authenticated
-        else settings.JITSI_CONFIGURATION.get("jitsi_guest_username"),
-        "email": user.email if user.is_authenticated else "",
-    }
+    video_grants = create_video_grants(room, is_admin, is_temp_room)
+    token_payload = api.AccessToken(
+        settings.LIVEKIT_CONFIGURATION["livekit_api_key"],
+        settings.LIVEKIT_CONFIGURATION["livekit_api_secret"],
```

Comment:
note: should automatically use the right env vars, according to [the doc](https://github.com/livekit/python-sdks?tab=readme-ov-file#generating-an-access-token)

If wrapped by django, we need to explicitly set them